pkg/extensions/search/cve/cve.go

```diff
@@ -1,205 +1,207 @@
 package cveinfo
 
 import (
-	"flag"
 	"fmt"
-	"path"
-	"strings"
-
-	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
-	"github.com/aquasecurity/trivy/pkg/commands/artifact"
-	"github.com/aquasecurity/trivy/pkg/commands/operation"
-	"github.com/aquasecurity/trivy/pkg/report"
-	"github.com/aquasecurity/trivy/pkg/types"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/opencontainers/go-digest"
 	ispec "github.com/opencontainers/image-spec/specs-go/v1"
-	"github.com/urfave/cli/v2"
 	"zotregistry.io/zot/pkg/extensions/search/common"
+	cvemodel "zotregistry.io/zot/pkg/extensions/search/cve/model"
+	"zotregistry.io/zot/pkg/extensions/search/cve/trivy"
 	"zotregistry.io/zot/pkg/log"
 	"zotregistry.io/zot/pkg/storage"
 )
 
-func getRoutePrefix(name string) string {
-	names := strings.SplitN(name, "/", 2) //nolint:gomnd
-
-	if len(names) != 2 { // nolint: gomnd
-		// it means route is of global storage e.g "centos:latest"
-		if len(names) == 1 {
-			return "/"
-		}
-	}
-
-	return fmt.Sprintf("/%s", names[0])
+type CveInfo interface {
+	GetImageListForCVE(repo, cveID string) ([]ImageInfoByCVE, error)
+	GetImageListWithCVEFixed(repo, cveID string) ([]common.TagInfo, error)
+	GetCVEListForImage(image string) (map[string]cvemodel.CVE, error)
+	GetCVESummaryForImage(image string) (ImageCVESummary, error)
+	UpdateDB() error
 }
 
-// UpdateCVEDb ...
-func UpdateCVEDb(dbDir string, log log.Logger) error {
-	return operation.DownloadDB("dev", dbDir, false, false, false)
+type Scanner interface {
```

Comment:
Good stuff and this becomes important if trivy stops being opensource or drops support